aws/accountid/mode/mode.go

```diff
@@ -0,0 +1,29 @@
+package mode
+
+import "fmt"
+
+// AIDMode switches on/off the account ID based endpoint routing
+type AIDMode string
+
+// enums of valid AIDMode
+const (
+	Preferred AIDMode = "preferred"
```

Comment:
We should have a none/unset sentinel i.e. `AccountIDEndpointModeUnset` (which is just the empty string).